client/src/pages/components/user/Dashboard.jsx

```diff
@@ -1,14 +1,119 @@
 import React from "react";
+import { useState } from "react";
 import { useEffect } from "react";
-import { useLocation } from "react-router";
+import { useNavigate } from "react-router";
 import AdminApi from "../../../api/AdminApi";
+import UserApi from "../../../api/UserApi";
+import avatar from "../../../assets/img/dummy.png";
+import moment from "moment";
 
 const Dashboard = () => {
+  const navigate = useNavigate();
+  if (!UserApi.isLogin) {
+    window.location = "/";
+  }
   if (AdminApi.isAdmin() === "admin@gmail.com") {
     window.location = "/admin/dashboard";
   }
 
-  return <div>home</div>;
+  const [user, setUser] = useState([]);
+  const [result, setResult] = useState([]);
+
+  useEffect(() => {
+    UserApi.getuser().then((res) => {
+      setUser(res.data.user[0]);
+      setResult(res.data.result);
+    });
+  }, []);
+
+  let lessonCount = result.length;
+
+  const [gscore, setGscore] = useState();
+  useEffect(() => {
+    let score = 0;
+    result.map((item, index) => {
+      item.score.filter((item) => {
+        if (item === true) {
+          score++;
+        }
+      });
+      setGscore(score);
+    });
+  }, [lessonCount]);
+
+  const wordsLearn = () => {
+    navigate("/words-learned", {
+      state: { user, result, gscore },
+    });
+  };
+
+  return (
+    <div>
+      <div className="container">
+        <div className="d-flex flex-100">
+          <div className="profile">
+            <h4>Dashboard</h4>
+            <div className="d-flex d-flex align-items-center justify-content-center">
+              <div className="avatar">
+                <img src={avatar} alt="..." className="rounded avatar" />
+              </div>
+              <div className="details">
+                <h5 className="name">{user.lastname + "," + user.firstname}</h5>
```

Comment:
```suggestion
                <h5 className="name">{`${user.lastname} , ${user.firstname}`}</h5>
```

for more readable approach